spec/index.bs

```diff
@@ -596,24 +596,36 @@ Issue(gpuweb/gpuweb#296): Consider merging all read-only usages.
 Textures may consist of separate [=mipmap levels=] and [=array layers=],
 which can be used differently at any given time.
 Each such <dfn dfn>subresource</dfn> is uniquely identified by a
-[=texture=], [=mipmap level=], and
-(for {{GPUTextureDimension/2d}} textures only) [=array layer=].
+[=texture=], [=mipmap level=],
+(for {{GPUTextureDimension/2d}} textures only) [=array layer=],
+and [=aspect=].
 
-The **main usage rule** is that any [=subresource=]
-at any given time can only be in either:
+<dfn dfn>atom resource</dfn> means a whole buffer or a single subresource of a texture.
```

Comment:
why are we introducing this definition (and/or renaming "subresource" into it)?

Reply:
"subresource" is limited to texture, but resource usage tracking and validation should include buffer too. Using "(texture) subresource" is inadequate, per my understanding. 

Well, we do have the concept of "subresource" of buffer (in Vulkan at least), but right WebGPU doesn't support that. And I didn't see any intention to support buffer subresource in WebGPU. Even in Vulkan, I don't know whether we can do buffer subresource tracking and validation.

Reply:
I think we should just fix the definition of a subresource instead of introducing an *additional* term to the spec. We can always disambiguate by saying "texture subresource" versus "buffer subresource" if the context demands it. In most contexts where a subresource is used, I expect it to not care about textures versus buffers.

Reply:
Hmm, if you search "subresource" in WebGPU spec, you can see that almost all "subresource" means "texture subresource", that's fine. They are actually talking about texture and texture view and so on. And I didn't replace those "subresource". 

But in the subsection "Resource Usages", I don't think "subresource" or "texture subresource" is appropriate. And I think we can either introduce a new concept (if "atom resource" is not good, we can rename it), or replace "subresource" with "a whole buffer and a single texture subresource" for resource usage tracking and validation. Otherwise, it is not clear. And it is misleading, considering that some native graphics API like Vulkan can support buffer subresource, but WebGPU can't so far, imho.

Reply:
The "subresource" used in the synchronization section totally doesn't care if it's a buffer versus texture.
The other uses of "subresource" appear to be clearly in the *context* of a texture, so it's not ambiguous.

What is your concern with just fixing the definition of a subresource to include whole buffers?

Reply:
we can define subresource and say that a subresource means a whole buffer or a slice of a texture. 

My concern is that subresource here is actually different from some native graphics API like Vulkan, whose buffer subresource means a part of a buffer, just like a part (mip slice, or array slice, or aspect) of a texture. It is consistent for both buffer and texture. However, buffer subresource is actually the entire buffer in WebGPU, which is literally not a "sub" resource for buffer. So I tend to use a new term. It is not that hard.

Reply:
that sounds good to me